Fix/regular

y4ney opened the pull request:
#  Inability to accurately extract keywords from the audit's output for comparison with flags
## Issue
https://github.com/aquasecurity/docker-bench/issues/108#issue-1846124623
## Step
1. Run the audit to see if there is a container that shares the user namespace.
```bash
$ docker ps --quiet --all | xargs docker inspect --format '{{ .Id }}:UsernsMode={{ .HostConfig.UsernsMode }}'
25bb4ff9b617445860b2db256789b6fc6974b563ec3fd162b4b5a2030b2a6dc3:UsernsMode=
c7536667b00fcd4e258ade2e7cea8412e838093b644905b60770e0557e8379a9:UsernsMode=host
```
2. Assign the output to `s` and pass it to the `getFlagValue`
```Bash
func getFlagValue(s, flag string) string {
  if flag == "" {
    return s
  }

  var flagVal string
  pttns := []string{
    `(?:^|[\s]+)"?` + flag + `"?\s*[=:][\r\t\f\v ]*"(.*)"`,
    `(?:^|[\s]+)"?` + flag + `"?\s*[=:][\r\t\f\v ]*([^\s]*)`,
    `(?:^|[\s]+)"?` + flag + `"?\s+([^-\s]+)`,
    `(?:^|[\s]+)` + `(` + flag + `)` + `(?:[\s]|$)`,
  }
  for _, pttn := range pttns {
    flagRe := regexp.MustCompile(pttn)
    vals := flagRe.FindStringSubmatch(s)
    for i, currentValue := range vals {
      if i == 0 {
        continue
      }
      if len(currentValue) > 0 {
        flagVal = currentValue
        return flagVal
      }
    }
  }
  return flagVal
}
```
3.It returned an empty string instead of a `host`.
## Modification
There may be no whitespace  before `flag`.
before
<img width="1068" alt="image" src="https://github.com/aquasecurity/bench-common/assets/44939500/7440dccd-f1d6-4a4e-bfcd-a0c318fe25c9">

after
<img width="1072" alt="image" src="https://github.com/aquasecurity/bench-common/assets/44939500/b95018fe-2440-4411-84c5-1de408c4802b">

y4ney committed:
modify regular

y4ney committed:
Merge branch 'aquasecurity:main' into fix/regular

chen-keinan commented:
@y4ney thank you for the contribution, can you please take a look at the failing test

y4ney committed:
The original rule has been restored and a new rule has been added.

y4ney committed:
add a test cases

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/aquasecurity/bench-common?pullRequest=121) <br/>All committers have signed the CLA.

y4ney commented:
Hi @chen-keinan , I have  restored the original rule and added a new rule.In order to ensure correctness, I also added a test case.I would appreciate it if you could review it again.

chen-keinan commented:
lgtm 🚀
